File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::map<int,int> _map;
        for (auto i = 0; i < nums.size(); i++) {
            if (auto val = _map.find(nums[i]); val != _map.end())
                return true;
            _map[nums[i]] = 1;
        }
        return false;
    }
};

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        # fast and slow pointers
        f, s = 0, 0
        while max(f,s) < len(nums):
            s = nums[s]
            f = nums[nums[f]]
            if s == f:
                break
        s = 0
        while s != f:
            s = nums[s]
            f = nums[f]
        return s